identity/scim.go

```diff
@@ -191,12 +227,27 @@ type UserList struct {
 	Resources    []ScimUser `json:"resources,omitempty"`
 }
 
+// ServicePrincipalList contains a list of ServicePrincipals fetched from a list api call from SCIM api
+type ServicePrincipalList struct {
+	TotalResults int32                  `json:"totalResults,omitempty"`
+	StartIndex   int32                  `json:"startIndex,omitempty"`
+	ItemsPerPage int32                  `json:"itemsPerPage,omitempty"`
+	Schemas      []URN                  `json:"schemas,omitempty"`
+	Resources    []ScimServicePrincipal `json:"resources,omitempty"`
+}
+
 // UserPatchRequest is a struct that contains all the information for a PATCH request to the SCIM users api
 type UserPatchRequest struct {
 	Schemas    []URN                 `json:"schemas,omitempty"`
 	Operations []UserPatchOperations `json:"Operations,omitempty"`
 }
 
+// ServicePrincipalPatchRequest is a struct that contains all the information for a PATCH request to the SCIM service principal api
+type ServicePrincipalPatchRequest struct {
```

Comment:
reuse UserPatchRequest, schema is 100% the same

Reply:
ad0a9e4065935cd6cda6064ce5ad45ca445f9bac